src/main/python/tests/examples/tutorials/test_adult.py

```diff
@@ -0,0 +1,324 @@
+
+        network = FFN_package.train(X, Y, 1, 16, 0.01, 1)
+
+        self.assertTrue(type(network) is not None) # sourcing and training seems to works
+
+        FFN_package.save_model(network, '"model/python_FFN/"').compute(verbose=True)
+
+        # TODO This does not work yet, not sure what the problem is
+        #probs = FFN_package.predict(Xt, network).compute(True)
+        # FFN_package.eval(Yt, Yt).compute()
+
+
+
+    def test_level1(self):
+        # Reduced because we want the tests to finish a bit faster.
+        train_count = 15000
+        test_count = 5000
+        train_data, train_labels, test_data, test_labels = self.d.get_preprocessed_dataset(interpolate=True,
+                                                                                           standardize=True, dimred=0.1)
+        # Train data
+        X = self.sds.from_numpy(train_data[:train_count])
+        Y = self.sds.from_numpy(train_labels[:train_count])
+        Y = Y + 1.0
+
+        # Test data
+        Xt = self.sds.from_numpy(test_data[:test_count])
+        Yt = self.sds.from_numpy(test_labels[:test_count])
+        Yt = Yt + 1.0
+
+        betas = multiLogReg(X, Y)
+
+        [_, y_pred, acc] = multiLogRegPredict(Xt, betas, Yt).compute()
+        self.assertGreater(acc, 80) #Todo remove?
+        # todo add text how high acc should be with this config
+
+        confusion_matrix_abs, _ = confusionMatrix(self.sds.from_numpy(y_pred), Yt).compute()
+        # todo print confusion matrix? Explain cm?
+        self.assertTrue(
+            np.allclose(
+                confusion_matrix_abs,
+                np.array([[3583, 502],
+                          [245, 670]])
+            )
+        )
+
+    def test_level2(self):
+
+        train_count = 32561
+        test_count = 16281
+
+        SCHEMA = '"DOUBLE,STRING,DOUBLE,STRING,DOUBLE,STRING,STRING,STRING,STRING,STRING,DOUBLE,DOUBLE,DOUBLE,STRING,STRING"'
+
+        F1 = self.sds.read(
+            self.dataset_path_train,
+            schema=SCHEMA
+        )
+        F2 = self.sds.read(
+            self.dataset_path_test,
+            schema=SCHEMA
+        )
+
+        jspec = self.sds.read(self.dataset_jspec, data_type="scalar", value_type="string")
+        PREPROCESS_package = self.sds.source(self.preprocess_src_path, "preprocess", print_imported_methods=True)
+
+        X1 = F1.rbind(F2)
+        X1, M1 = X1.transform_encode(spec=jspec)
+
+        X = PREPROCESS_package.get_X(X1, 1, train_count)
+        Y = PREPROCESS_package.get_Y(X1, 1, train_count)
+
+        Xt = PREPROCESS_package.get_X(X1, train_count, train_count+test_count)
+        Yt = PREPROCESS_package.get_Y(X1, train_count, train_count+test_count)
+
+        Yt = PREPROCESS_package.replace_value(Yt, 3.0, 1.0)
+        Yt = PREPROCESS_package.replace_value(Yt, 4.0, 2.0)
+
+        # better alternative for encoding. This was intended, but it does not work
```

Comment:
still the case?

Reply:
Yes. The replace function you made for us to clean the target labels works for one label correctly, however not for the other (See our discussion that started on 5.7.).
The code that is commented out is how it should be done, using the replace function and the transform_apply function. The not-commented code is a temporary, working solution, but it is not optimal.